### PR TITLE
[gdal] Fix gdal JPEG driver on iOS

### PR DESCRIPTION
**Pull request by nirvn**
This PR fixes the gdal JPEG driver on iOS by relying on gdal's internal jpeg library instead of the default external library. The issue was detected by our QField iOS users who noticed JPEG raster datasets (as well as JPEG encoded geotiffs, mbtiles, etc.) would not render on their mobile device.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by nirvn** [comment]
@microsoft-github-policy-service agree

File: ports/gdal/portfile.cmake [hunk 1]
```diff
@@ -57,6 +57,11 @@ if(GDAL_USE_ICONV AND VCPKG_TARGET_IS_WINDOWS)
     list(APPEND FEATURE_OPTIONS -D_ICONV_SECOND_ARGUMENT_IS_NOT_CONST=ON)
 endif()
 
+# JPEG driver (et cie) on iOS doesn't work when built against external lib
+if(VCPKG_TARGET_IS_IOS AND GDAL_USE_JPEG)
+    list(APPEND FEATURE_OPTIONS -DGDAL_USE_JPEG_INTERNAL=ON)
+endif()
```
Comment (dg0yt) [inline ports/gdal/portfile.cmake:63]: So, "doesn't work"...
This looks like a workaround, not like a bug fix. 
Did you check build logs for the relevant drivers? 
Did you try to contact upstream for known issues with regard to an external jpeg lib on iOS?
(How do we know that it is also broken for other users, not just a particular QField build?)

Comment (nirvn) [reply ports/gdal/portfile.cmake:63]: @dg0yt , fair questions, and thanks for the blazing fast responsiveness :)

>> So, "doesn't work"...

Correct, gdal simply fails to open JPEG raster dataset, and fails to render other raster types that rely on JPEG compression.

>>  This looks like a workaround, not like a bug fix.

Yeah, that's a more accurate word. It mimics the workaround applied for GDAL_USE_SHAPELIB[_INTERNAL]. 

>>  Did you check build logs for the relevant drivers?

I'm not sure I understand that specific question; the JPEG driver relies on libjpeg-turbo when building against an external library on vcpkg.

>>  Did you try to contact upstream for known issues with regard to an external jpeg lib on iOS?

No, I have not. It's unclear to me whether the root cause/issue lies with libjpeg-turbo or gdal. 

>> How do we know that it is also broken for other users, not just a particular QField build?

We use vcpkg to build QField on Android, iOS, and linux using essentially the same recipe, and while Android and linux builds have no problem with gdal loading JPG raster datasets, iOS fails. 

I've tried to look around for another open source project building gdal on iOS via vcpkg but couldn't find any, so the only pool of builds/testers I'm aware of are QField users. It must be said too that the issue has been reported for a long time, across gdal at least (out of memory) 3.5, 3.6, and 3.7.

Comment (dg0yt) [reply ports/gdal/portfile.cmake:63]: > It mimics/is inspired from the workaround applied for GDAL_USE_SHAPELIB[_INTERNAL].

This isn't a good example. Using an external shapelib is generally discouraged. It is not a vcpkg issue. https://github.com/OSGeo/gdal/issues/5711

> I've tried to look around for another open source project building gdal on iOS via vcpkg but couldn't find any.

That's why I point at upstream. I want to know if there is a general problem (upstream issue) before applying a workaround here which is only tested with one particular downstream.

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by nirvn** [comment]
Alright, closing this for now as an issue filed upstream got me to rethink this and figure out what the issue at play is here (see https://github.com/OSGeo/gdal/issues/7904#issuecomment-1578656103)

@dg0yt , @LilyWangLL , thanks to both of you for your responsiveness.

**Pull request closed without merging**